docs/spec/resources.md

```diff
@@ -42,6 +42,29 @@ resource dnsZone 'Microsoft.Network/dnszones@2018-05-01' = {
 }
 ```
 
+## Resource nesting
+
+A resource declaration may appear inside another resource declaration when the inner resource is a child type of the other type.
+
+```
+resource myParent 'My.Rp/parentType@2020-01-01' = {
+  name: 'myParent'
+  location: 'West US'
+
+  // declares a resource of type 'My.Rp/parentType/childType@2020-01-01' 
+  resource myChild 'childType' = {
+    name: 'myChild'
+    properties: {
+      displayName: 'child in ${parent.location}'
+    }
+  }
+}
+```
+
+When used in this form the nested declaration must use a simple `name` with a single segment.
+
+A nested resource declaration must appear at the top level of syntax of the containing resource. Declarations may be nested arbirarily deep, as long as each level is a child type of its containing resource. A nested resource may access properties of its containing resource. A containing resource may not access properties of the resources it contains, this would cause a cyclic-dependency.
```

Comment:
I think it would be good to explain here how the `type` string is interpreted (just a single segment provided, API version inferred from parent if not provided).

Reply:
Sure will add. I was trying to keep it terse since the rest of the docs are. **I SHALL OPEN THE FLOODGATES**